js_code/offerList.js

```diff
@@ -0,0 +1,528 @@
+            hashtags : ['PC', 'CS', 's1mple', 'MacBook Pro', 'DELL'],
+            discount : '70',
+            validUntil : new Date('2022-01-21T20:12:32'),
+            rating : 5,
+            reviews : ['Xbox is better!'],
+        },
+        {
+            id: '20',
+            description : 'Компьютеры со скидкой 88%',
+            createdAt : new Date('2021-02-11T20:14:32'),
+            link : 'https://PC.pl',
+            vendor : 'Computers - best',
+            photoLink : 'https://lh3.googleusercontent.com/proxy/n4y3C7gnmGV4SOVmkE2DoLGua_DNkt9Nyrnth8-gXnbO_xFewvtRi3ztskcGA4G1mwE45gvXpq_1rWX1VYLXJbh9jJGUc3x0E6exe2T9F1LLN5ZgL4d8KZy4SoaGo9IlVQGFY6mDTznWSptH9XZIf-lK',
+            hashtags : ['PC', 'CS', 's1mple', 'MacBook Pro', 'DELL'] ,
+            discount : '88',
+            validUntil : new Date('2022-01-18T22:19:42'),
+            rating : 1,
+            reviews : ['Xbox is better!'] ,
+        }
+    ]
+
+  function getAds(skip = 0, top = 10, filterConfig){
+    if (typeof skip !== 'number' || typeof top !== 'number') {
+        console.log('Error with inputting types!');
+        return;
+    }
+
+    let returningAdsList = adList;
+
+    if (filterConfig){
+        for (let param in filterConfig){
+            if (param === 'hashtags'){
+                for (let i = 0; i < filterConfig.hashtags.length; i++){
+                    returningAdsList = returningAdsList.filter(item => item.hashtags.includes(filterConfig.hashtags[i]));
+                }
+            }
+            else if (param === 'vendor'){
+                returningAdsList = returningAdsList.filter(item => item.vendor === filterConfig.vendor);
+            }
+            else if (param === 'dateFrom'){
+                returningAdsList = returningAdsList.filter(item => item.createdAt >= filterConfig.dateFrom);
+            }
+            else if (param === 'dateTo'){
+                returningAdsList = returningAdsList.filter(item => item.createdAt < filterConfig.dateTo);
+            }
+        }
+    }
+
+    returningAdsList.sort(comparator);
+    return returningAdsList.slice(skip, skip + top);
+
+    function comparator(first, second) {
+        if (first.createdAt < second.createdAt){
+            return 1;
+        }
+        else if (first.createdAt > second.createdAt){
+            return -1;
+        }
+        else{
+            return 0;
+        }
+    }
+  }
+
+function getAd(id){
+    if (typeof id === 'string'){
+        return adList.find(item => item.id == id);
+    }
+    else{
+        console.log('Incorrect type of id!')
+    }
+}
+
+function validateAd(adItem){
+    let parameters = ['id', 'description',  'createdAt', 'link', 'vendor', 'photoLink', 'validUntil', 'discount', 'hashtags'];
+    for(let i = 0; i < parameters.length; i++){
+        if(adItem[parameters[i]] == undefined){
+            console.log('pososi');
```

Comment:
Do you need this comment here? 🤨

Reply:
I removed the comments.
Excuse me please! I used them as a test of how functions work ..
This is a good experience for me. Now I will adhere to a business style regardless of work. I hope you will understand me and this will not affect the current and future assessments.